src/Microsoft.ServiceFabric.Services.Queryable/Service/ReliableStateExtensions.cs

```diff
@@ -64,58 +69,183 @@ internal static class ReliableStateExtensions
-		private static async Task<IReliableState> GetQueryableState(this IReliableStateManager stateManager, string collection)
+	    /// <summary>
+	    /// Delete from the reliable collection with the given key from the reliable state manager using the given parameters.
+	    /// </summary>
+	    /// <param name="stateManager">Reliable state manager for the replica.</param>
+	    /// <param name="collection">Name of the reliable collection.</param>
+	    /// <param name="keyJson">Entity Key.</param>
+	    /// <returns>A boolean value based on the success of the operation.</returns>
+        public static async Task<bool> DeleteAsync(this IReliableStateManager stateManager, string collection,
+	        string keyJson)
+	    {
+
+	        var dictionary = await stateManager.GetQueryableState(collection).ConfigureAwait(false);
+	        try
+	        {
+	            using (ITransaction tx = stateManager.CreateTransaction())
+	            {
+	                var keyType = dictionary.GetKeyType();
+	                var valueType = dictionary.GetValueType();
+	                var key = JsonConvert.DeserializeObject(keyJson, keyType);
+	                var dictionaryType = typeof(IReliableDictionary<,>).MakeGenericType(keyType, valueType);
+	                await (Task) dictionaryType.GetMethod("TryRemoveAsync", new[] {typeof(ITransaction), keyType})
+	                    .Invoke(dictionary, new object[] {tx, key});
+
+	                await tx.CommitAsync();
+                
+	            }
+	        }
+	        catch (ArgumentException)
+	        {
+
+	            throw new HttpException(400, $"A value with given key:{keyJson} does not exist.");
+
+	        }
+
+	        return true;
+
+	    }
+
+        /// <summary>
+        /// Add to the reliable collection the given key & value using the reliable state managers.
+        /// </summary>
+        /// <param name="stateManager">Reliable state manager for the replica.</param>
+        /// <param name="collection">Name of the reliable collection.</param>
+        /// <param name="keyJson">Entity Key.</param>
+        /// <param name="valJson">Value.</param>
+        /// <returns>A boolean value based on the success of the operation.</returns>
+        public static async Task<bool> AddAsync(this IReliableStateManager stateManager, string collection, string keyJson,
+	        string valJson)
+	    {
+
+	        var dictionary = await stateManager.GetQueryableState(collection).ConfigureAwait(false);
+
+	        try
+	        {
+	            using (ITransaction tx = stateManager.CreateTransaction())
+	            {
+	                var keyType = dictionary.GetKeyType();
+	                var valueType = dictionary.GetValueType();
+	                var key = JsonConvert.DeserializeObject(keyJson, keyType);
+	                var val = JsonConvert.DeserializeObject(valJson, valueType);
+
+	                var dictionaryType = typeof(IReliableDictionary<,>).MakeGenericType(keyType, valueType);
```

Comment:
refactor this code - create a private static method to get the dictionary type from dictionary instance.